### PR TITLE
fix(amazon q): add pl/1 and bms extensions as a recognized code file type

### PR DESCRIPTION
**Pull request by scottwritescode**
## Problem
The codefileExtensions does not contain an extension type for `.pli` which is the common extension for PL/1, or for `.bms` which is Basic Mapping Support and common for screen definition files in IBM mainframe. Because of this, the files and code in them are not accessible in the workspace context for Amazon Q using the `/dev` quick action.

When using the Q quick action '\dev' for `.pli` or `.bms` files, Q responds with the following message
> This appears to be an empty program or workspace with no source files present. There is nothing to explain at this time since no code or files are available for analysis.

## Solution
- Add the PL/1 language extension `.pli` as a known code file extension type
- Add the Basic Mapping Support extension `.bms` as a known code file extension type

No tests were updated as there are none exclusively for filetype extensions

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by justinmk3** [comment]
Thanks for your help here. These filetypes are decided by the Q service team when they feel like the service is ready to handle particular cases. However, I have forwarded this to them.

File: packages/amazonq/.changes/next-release/Bug Fix-80f16b7a-440d-4e06-89ad-1454c58ff28f.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
+	"description": "Q feature dev: update file extension list"
```
Comment (justinmk3) [inline packages/amazonq/.changes/next-release/Bug Fix-80f16b7a-440d-4e06-89ad-1454c58ff28f.json:3]:
```suggestion
	"description": "Q feature dev: recognize .bms, .pli code files"
```

File: packages/amazonq/.changes/next-release/Bug Fix-80f16b7a-440d-4e06-89ad-1454c58ff28f.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Bug Fix",
```
Comment (willyyhuang) [inline packages/amazonq/.changes/next-release/Bug Fix-80f16b7a-440d-4e06-89ad-1454c58ff28f.json:2]:
```suggestion
	"type": "Feature",
```

Thank you for contributing! This is not exactly a "bug" per say, it is by design

File: packages/core/src/shared/filetypes.ts [hunk 3]
```diff
@@ -161,6 +161,7 @@ export const codefileExtensions = new Set([
     '.bash',
     '.bat',
     '.boo',
+    '.bms',
```
Comment (willyyhuang) [inline packages/core/src/shared/filetypes.ts:164]: nit: this is not sorting alphabetically. but not a blocker

**Review by willyyhuang — APPROVED** [review]

**Pull request merged**